lib/src/dom/pretty_dom.dart

```diff
@@ -70,9 +68,9 @@ String prettyDOM(
 
 @JS('rtl.prettyDOM')
 external String _jsPrettyDOM([
-  Node dom,
-  int maxLength,
-  PrettyDomOptions options,
+  Node? dom,
+  int? maxLength,
+  PrettyDomOptions? options,
```

Comment:
I'm basing this off these being optional and nullable on the JS side - I'm not really sure the best way to handle a case where something should be nullable, but the JS side might not actually handle Dart null well. @greglittlefield-wf have you run into this before?

In this case, null `node` does work so I'll add a test for that. And I guess this function is private so it matters more how this is handled in the Dart wrapper

Reply:
I think this case will come up a lot with RMUI props where the prop is optional and therefore nullable, but the JS component doesn't handle Dart null being passed in

Reply:
Yeah there may be some cases where JS handles `undefined`, but not `null`; it really depends on the function.

When we're talking about optional JS interop arguments, it also depends on how it's being called.

I believe `_jsPrettyDOM(node)` is equivalent to omitting those args in JS, but not equivalent to `_jsPrettyDOM(node, null)` which explicitly passes `null` to the second arg, and may or may not be handled correctly on the JS side.

I haven't run into this too much, but yeah I don't think there's really a great way to handle this, other than creating multiple "overloaded" versions of the wrapper to ensure `null` isn't passed into positional arguments that are both optional and non-nullable. E.g.,
```dart
@JS('rtl.prettyDOM')
external String _jsPrettyDOM0();
@JS('rtl.prettyDOM')
external String _jsPrettyDOM1(Node dom);
@JS('rtl.prettyDOM')
external String _jsPrettyDOM2(Node dom, int maxLength);
@JS('rtl.prettyDOM')
external String _jsPrettyDOM3(Node dom, int maxLength, PrettyDomOptions options);
```

> And I guess this function is private so it matters more how this is handled in the Dart wrapper

+1; if these aren't publicly exposed and we have test coverage around the cases, then we can type the interop whatever we want.

Reply:
Also FYI based on the JS typing [in the docs](https://testing-library.com/docs/dom-testing-library/api-debugging/#prettydom), it looks like the `node` arg is required

Reply:
Oh weird - I was looking at [this](https://github.com/testing-library/dom-testing-library/blob/a7b725257e61370675ee043813003d36fae23e9d/types/pretty-dom.d.ts#L12)

Reply:
>I think this case will come up a lot with RMUI props where the prop is optional and therefore nullable, but the JS component doesn't handle Dart null being passed in

Mm that's a good point, I'll have to think on that a bit... Like, them explicitly passing `null` in I'm not too worried about, since if it caused issues/exceptions that would definitely get caught during development. But passing in a nullable value that's only sometimes null... Gross

Reply:
> Oh weird - I was looking at [this](https://github.com/testing-library/dom-testing-library/blob/a7b725257e61370675ee043813003d36fae23e9d/types/pretty-dom.d.ts#L12)

Huh yeah they're probably just out of sync. I'm fine keeping it as optional/nullable!